### PR TITLE
HDDS-10637. Add config for whitelisting X509 signature algorithms

### PR DESCRIPTION
**Pull request by Galsza**
## What changes were proposed in this pull request?
Create a configuration option that can be later used in a compliance suite to filter available X509 signature algorithm.

This whitelist option will check if the provided algorithm in the base config option is compliant with it. However this check will be performed in [HDDS-10604 Whitelist based compliance check for crypto related configuration options
](https://issues.apache.org/jira/browse/HDDS-10604) This issue is part of an ongoing effort to create a whitelist option for all security related configurations.

## Link to the apache jira
[HDDS-10637](https://issues.apache.org/jira/browse/HDDS-10637)

## How was this patch tested?

No functional change yet, only an empty config option. Green ci:
https://github.com/Galsza/ozone/actions/runs/8525063840

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java [hunk 1]
```diff
@@ -682,6 +682,9 @@ public final class OzoneConfigKeys {
   public static final String
       OZONE_OM_NETWORK_TOPOLOGY_REFRESH_DURATION_DEFAULT = "1h";
 
+  public static final String
+      OZONE_COMPLIANCE_ALLOWED_HDDS_X509_SIGNATURE_ALGO = "ozone.compliance.allowed.hdds.x509.signature.algorithm";
+  public static final String OZONE_COMPLIANCE_ALLOWED_HDDS_X509_SIGNATURE_ALGO_DEFAULT = "*";
```
Comment (dombizita) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java:687]: `HDDS_X509_SIGNATURE_ALGO` is defined in the `HddsConfigKeys` class, shouldn't we have this there too, right after that constant?

Comment (Galsza) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java]: Thank you Zita, you are right I've moved them. Initially I wanted all these config options to land at the end of OzoneConfigKeys to track them but the prefix is a better way to do it.

**Review by dombizita — APPROVED** [review]

**Review by fapifta — CHANGES_REQUESTED** [review]
Please refer to [this comment](https://github.com/apache/ozone/pull/6476#pullrequestreview-1988688328) under #6476

**Comment by Galsza** [comment]
@fapifta Thanks for the review, I've changed the PR. It's much shorter now.

**Pull request closed without merging**